python/cuml/experimental/common/base.pyx

```diff
@@ -385,15 +438,29 @@ class Base(TagsMixin,
         self.__dict__.update(d)
 
     def __getattr__(self, attr):
-        """
-        Redirects to `solver_model` if the attribute exists.
-        """
+        # Attempts to redirect calls to CPU estimator
+        if inspect.isclass(self):
```

Comment:
I'm generally a bit lost on this one. Can `__getattr__` ever be called as `classmethod` or `staticmethod` like this?

Reply:
> I'm generally a bit lost on this one. Can __getattr__ ever be called as classmethod or staticmethod like this?

If the method doesn't have an implementation in cuML, it will look for it in the CPU estimator. Yes, it indeed appears to be working fine with functions/methods.

> Could you add a comment explaining the purpose of this check?

Yes, but to explain it quickly, the `hasattr` calls would create an infinite recursion if done before class instantiation.

Reply:
But doesn't that mean it only works with classes? Why would `__getattr__` ever be called as `classmethod` outside of a metaclass scenario? Is that really intended?

Reply:
In the end, I removed the CPU redirection because it wasn't really part of the project (as discussed with Dante). But, it's still possible to reintroduce it if we change our minds.